backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/ElasticsearchIdProjection.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.search.projection.impl;
+
+import org.hibernate.search.backend.elasticsearch.search.impl.ElasticsearchSearchContext;
+import org.hibernate.search.engine.backend.common.DocumentReference;
+import org.hibernate.search.engine.search.loading.spi.LoadingResult;
+import org.hibernate.search.engine.search.loading.spi.ProjectionHitMapper;
+import org.hibernate.search.engine.search.projection.SearchProjection;
+import org.hibernate.search.engine.search.projection.spi.IdProjectionBuilder;
+
+import com.google.gson.JsonObject;
+
+public class ElasticsearchIdProjection<I> extends AbstractElasticsearchProjection<DocumentReference, I> {
+
+	private final DocumentReferenceExtractionHelper helper;
```

Comment:
I'm not sure that the document ids are enough to load the entity ids. I think that we still need `DocumentReference`s.

Reply:
Let's agree to disagree. As I said, have a look at how it's done for ID predicates.

All you need to convert a document ID to an entity ID is a converter, which will be backed by `org.hibernate.search.mapper.pojo.bridge.IdentifierBridge#fromDocumentIdentifier`. The type name is not involved there.

You have an example of how it can work with the `IdDslConverter`, which does the opposite job (converts from entity ID to document ID):

* At mapping time, `org.hibernate.search.engine.backend.document.model.dsl.spi.IndexSchemaRootNodeBuilder#idDslConverter` allows the mapper to register an ID converter for a given mapped type.
* The implementation of that converter for the POJO mapper is `org.hibernate.search.mapper.pojo.bridge.runtime.impl.PojoIdentifierBridgeToDocumentIdentifierValueConverter`, and it's backed by an `IdentifierBridge`.
* When creating an ID predicate, the identifier is retrieved in `org.hibernate.search.backend.elasticsearch.search.predicate.impl.ElasticsearchMatchIdPredicate.Builder#value`
* As you can see, `org.hibernate.search.backend.elasticsearch.scope.model.impl.ElasticsearchScopeSearchIndexesContext#idDslConverter` retrieves the converter for the targeted indexes, checking that all targeted indexes have the same converter.

Reply:
you're right!